runtime/jcl/common/criusupport.c

```diff
@@ -44,3 +54,94 @@ Java_com_ibm_jvm_CRIUSupport_isCRIUSupportEnabledImpl(JNIEnv *env, jclass unused
 
 	return res;
 }
+
+jint JNICALL
+Java_com_ibm_jvm_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
+							jclass unused,
+							jobject checkPointDir,
+							jboolean keepRunning,
+							jboolean shellJob,
+							jboolean extUnixSupport,
+							jint logLevel,
+							jobject logFile)
+{
+	jint rc = UNSUPPORTED_OPERATION;
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	char directoryBuf[STRING_BUFFER_SIZE] = {0};
+	char *directoryUTF = NULL;
+	char logFileBuf[STRING_BUFFER_SIZE] = {0};
+	char *logFileUTF = NULL;
+
+	if (J9_ARE_ALL_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_CRIU_SUPPORT)) {
+#if defined(LINUX)
+		j9object_t cpDir = NULL;
+		j9object_t log = NULL;
+		IDATA dirFD = 0;
+		PORT_ACCESS_FROM_VMC(currentThread);
+
+		vmFuncs->internalEnterVMFromJNI(currentThread);
+
+		cpDir = J9_JNI_UNWRAP_REFERENCE(checkPointDir);
+		log = J9_JNI_UNWRAP_REFERENCE(logFile);
+
+		directoryUTF = vmFuncs->copyStringToUTF8WithMemAlloc(currentThread, cpDir, J9_STR_NULL_TERMINATE_RESULT, "", 0, directoryBuf, STRING_BUFFER_SIZE, NULL);
+		if (NULL == directoryUTF) {
+			vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+			goto release;
+		}
+
+		logFileUTF = vmFuncs->copyStringToUTF8WithMemAlloc(currentThread, log, J9_STR_NULL_TERMINATE_RESULT, "", 0, logFileBuf, STRING_BUFFER_SIZE, NULL);
+		if (NULL == logFileUTF) {
+			vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+			goto freeDir;
+		}
+
+		dirFD = open(directoryUTF, O_DIRECTORY);
+		if (dirFD < 0) {
+			rc = INVALID_ARGUEMENTS;
+			goto freeLog;
+		}
+
```

Comment:
Does there need to be a call to `criu_init_opts` first?  

Asking based on
> Note: it should be called before using any other functions from libcriu, except criu_check(). Also you should use it to reinitialize options. It returns 0 on success and -1 on fail.

https://criu.org/C_API

Reply:
What will ensure that `criu_init_opts()` gets called exactly once?

Reply:
Where in the minimal spec does it say it should be called exactly once?  `Also you should use it to reinitialize options` implies it can be called anytime you want to reinit the options

Reply:
As you quoted from the documentation is 'reinitializes'. If some other part of the application has already called `criu_init_opts()` and then made other changes, our call to `criu_init_opts()` might undo some of that work. At least this sort of interaction should be documented.